### PR TITLE
Deploy remote collectd and collector on the monitoring node(s)

### PR DESCRIPTION
**Pull request by simonpasquier**
This depends on https://github.com/tcpcloud/salt-formula-collectd/pull/22

### CODE REVIEW

File: classes/system/collectd/remote_client/single.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+classes:
+- system.collectd.remote_client.output.heka
+parameters:
+  collectd:
+    remote_client:
+        automatic_starting: true
```
Comment (elemoine) [inline classes/system/collectd/remote_client/single.yml]: Two-space indent?

File: classes/system/heka/aggregator/cluster.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+classes:
+- service.heka.aggregator.cluster
+parameters:
+  heka:
+    aggregator:
```
Comment (SwannCroiset) [inline classes/system/heka/aggregator/cluster.yml]: please, add nagios_* params like it's done with the single.yml

Comment (simonpasquier) [inline classes/system/heka/aggregator/cluster.yml]: done

File: scripts/stacklight_monitor_install.sh [hunk 3]
```diff
@@ -1,7 +1,19 @@
 #!/bin/bash -x
 
 # StackLight services
-salt -C 'I@elasticsearch:server' state.sls elasticsearch -b 1
+salt -C 'I@elasticsearch:server' state.sls elasticsearch.server -b 1
 salt -C 'I@influxdb:server' state.sls influxdb -b 1
-salt -C 'I@kibana:server' state.sls kibana
-salt -C 'I@grafana:server' state.sls grafana
+salt -C 'I@kibana:server' state.sls kibana.server
+salt -C 'I@grafana:server' state.sls grafana.server -b 1
```
Comment (SwannCroiset) [inline scripts/stacklight_monitor_install.sh]: just `grafana` should be fine w/ https://github.com/tcpcloud/salt-formula-kibana/pull/3

Comment (thouveng) [inline scripts/stacklight_monitor_install.sh]: just `kibana` you mean ;-). But I think that it is easier to keep the same pattern and specify .server and .client everywhere.

Comment (simonpasquier) [inline scripts/stacklight_monitor_install.sh]: @SwannCroiset the idea is to be consistent with the other scripts (eg openstack_control_install.sh).

Comment (simonpasquier) [inline scripts/stacklight_monitor_install.sh]: Also it probably avoids race conditions to separate the execution of the server and client states.

Comment (SwannCroiset) [inline scripts/stacklight_monitor_install.sh]: ok thank for explanations, was just a comment anyway

File: scripts/stacklight_infra_install.sh [hunk 4]
```diff
@@ -14,6 +14,11 @@ sleep 5
 
 # Update monitoring node with mine metadata
 salt -C 'I@heka:aggregator' state.sls collectd,heka
+# Start manually the services that are bound to the monitoring VIP
+salt -G 'ipv4:172.16.10.253' system.start remote_collectd
+salt -G 'ipv4:172.16.10.253' system.start remote_collector
+salt -G 'ipv4:172.16.10.253' system.start aggregator
```
Comment (SwannCroiset) [inline scripts/stacklight_infra_install.sh]: you can add nagios as well here

File: scripts/stacklight_infra_install.sh [hunk 5]
```diff
@@ -14,6 +14,11 @@ sleep 5
 
 # Update monitoring node with mine metadata
 salt -C 'I@heka:aggregator' state.sls collectd,heka
+# Start manually the services that are bound to the monitoring VIP
+salt -G 'ipv4:172.16.10.253' system.start remote_collectd
```
Comment (SwannCroiset) [inline scripts/stacklight_infra_install.sh]: s/system/service/

Comment (simonpasquier) [inline scripts/stacklight_infra_install.sh]: done

**Pull request merged**